swip/common/paths.py

```diff
@@ -0,0 +1,41 @@
+from pathlib import Path
+
+from common.exceptions import NoSwipError
+
+
+def find_swip_repo(given_path):  
+    path = Path(given_path)
+    swip_path = None
+    if path.is_file():
+        path = path.parent
+
+    while path != path.parent and swip_path is None:
+        subdirs = [p for p in path.iterdir() if p.is_dir()] 
```

Comment:
It looks like generator expression fit here better.

Also, consider using `filter` instead: `filter(Path.is_dir, path.iterdir())`